sw/device/tests/rv_timer_test.c

```diff
@@ -11,18 +11,27 @@
 #include "sw/device/lib/pinmux.h"
 #include "sw/device/lib/runtime/check.h"
 #include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/ibex.h"
 #include "sw/device/lib/testing/test_main.h"
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 
 static dif_gpio_t gpio;
-// Flag to indicate that the interrupt was seen. Declared as volatile since it
-// is referenced in the ISR routine as well as in the main program flow.
-static volatile bool intr_handling_success;
-static const uint32_t hart = (uint32_t)kTopEarlgreyPlicTargetIbex0;
+static dif_rv_timer_t timer;
 
-bool test_main(void) {
-  const uint64_t cmp = 0x000000000000000F;
+// Flag for checking whether the interrupt handler was called. When the handler
+// is entered, this value *must* be set to false, to catch false positives.
+//
+// This variable is volatile, since it may suddenly change from the compiler's
+// perspective (e.g., due to an interrupt firing).
+static volatile bool irq_fired = true;
+
+static const uint32_t kHart = (uint32_t)kTopEarlgreyPlicTargetIbex0;
```

Comment:
Let's leave a TODO to ensure we catch the hart ids somewhere (as these may not correspond to the rv_plic targets).

Reply:
Could you elaborate on what you mean here?

Reply:
PLIC Target 0 does not necessarily mean Hart 0. I'm not sure that's a 1:1 mapping, though it would be good to make sure they do match up if we can. Maybe I'll file something against topgen to ensure this is the case, though I'm not sure `rv_timer.hjson` is generated from topgen yet.

Reply:
Hmm. I think this was originally `0` and then someone came along and changed it to the topgen constant. I'll add a comment and we can loop back on this later.